db/CSV_Writers/MISC/writer.js

```diff
@@ -0,0 +1,23 @@
+module.exports = (writer, str) => {
+  if (!writer.write(str)) {
+    return new Promise((resolve) => {
+      writer.once('drain', resolve);
+    });
+  };
+}
```

Comment:
is this being used somewhere? i didnt see drain used when you were writing your csv's but maybe i missed something :)

Reply:
No its not but I'm currently saving some unused nuggets i'm afraid to delete in the MISC file. I basically copied this from a stack overflow, and after reading your comments in slack it makes a little more sense.


The one thing that is a little confusing is that there is no else statement- is it calling writer.write(str) on line 2 and if that call returns false then its going to drain then resolve whereas if it returns true when there isn't an issue but it still runs regardless?